core/src/main/java/com/radixdlt/messaging/core/MessageDispatcher.java

```diff
@@ -137,26 +136,77 @@ CompletableFuture<Result<Unit, Cause>> send(final OutboundMessageEvent outboundM
         .findOrCreateChannel(outboundMessage.receiver())
         .thenApply(channel -> send(channel, bytes))
         .thenApply(this::updateStatistics)
-        .exceptionally(t -> completionException(t, receiver, message));
+        .thenApply(
+            result -> {
+              result.onError(err -> logSendError(message, receiver, err.toString()));
+              return result;
+            })
+        .exceptionally(
+            t -> {
+              final var cause = t.getCause() != null ? t.getCause().getMessage() : t.getMessage();
+              logSendError(message, receiver, cause);
+              return IO_ERROR.result();
+            });
+  }
+
+  private void logSendError(Message message, NodeId receiver, String cause) {
+    synchronized (sendErrorLogRateLimitersByReceiver) {
+      final RejectionCountingRateLimiter rateLimiter;
+      if (sendErrorLogRateLimitersByReceiver.contains(receiver)) {
+        rateLimiter = sendErrorLogRateLimitersByReceiver.get(receiver).orElseThrow();
+      } else {
+        rateLimiter = new RejectionCountingRateLimiter(0.05);
+        sendErrorLogRateLimitersByReceiver.put(receiver, rateLimiter);
+      }
+      rateLimiter.tryAcquire(
+          countSinceLastPermit -> {
+            final var baseMsg =
+                String.format(
+                    "An outbound message of type %s couldn't be send to %s because of: \"%s\".",
+                    message.getClass().getSimpleName(),
+                    addressing.encodeNodeAddress(receiver.getPublicKey()),
+                    cause);
+            if (countSinceLastPermit > 0) {
+              log.warn(
+                  "{} {} more messages couldn't be send to this peer since "
+                      + "the previous log message (likely for the same reason).",
+                  baseMsg,
+                  countSinceLastPermit);
+            } else {
+              log.warn(baseMsg);
+            }
+          });
+    }
+  }
+
+  private void logTtlExpired(Message message, NodeId receiver) {
+    synchronized (ttlExpiredLogRateLimiter) {
+      ttlExpiredLogRateLimiter.tryAcquire(
+          countSinceLastPermit -> {
+            final var baseMsg =
+                String.format(
+                    "TTL (of %s ms) has expired for an outbound message of type %s destined to %s.",
```

Comment:
If I read this message I wouldn't understand:
* Why might that be?
* What is the consequence?

Could we explain EG that the consequence is that the message is dropped? And (if it's true) that the Why is likely that there there is a message sending backlog?

Reply:
This seems like quite a scary proposition - out of interest - do we have a metric for the backlog of messages in our outbound queue?

Reply:
we do have a metric: `metrics.messages().outbound().queued()`